### ISSUE TITLE
Stackoverflow when not using defaultExpanded

### ISSUE THREAD

**Issue by andre-byrne** [hirohe/storybook-addon-jarle-monaco#2]
This is an amazing add-on! My docs have never looked better, and folks at my company are very excited about the idea of directly editing the components rather than fiddling with knobs. Truly amazing!

That being said, I've found an issue. It's not a show stopper, but it is awkward:

**Problem:**

> If you have a mix of editors that require horizontal scrolling and editors that don't, and they are not expanded by default, then the editors crash with a stack overflow. It seems `setScrollDimensions` in some dependency is being called endlessly. This only appears to happen on initial page load. If I load up some editors that have no horizontal scrolling, and then shrink the screen down so that they do need it, they do not crash.

**Reproduction Steps:**

In your example repo I've modified the following files:

 - to the mdx live edit example, add a third playground that is just the first example again
 - in the components file for that mdx story, remove `defaultExpanded` from all the demos
 - in Demo2, remove all the line breaks so that the editor will require scrolling

Here are the files, for your copy/paste convenience.

```ts
// stories/MarkdownDemo.stories.mdx
import { Meta } from '@storybook/addon-docs';
import { Demo1, Demo2, Demo3 } from './markdown-demo';

<Meta title="Example/LiveEdit in MDX" />

<style>{`
.playground {
  margin-bottom: 24px;
}
`}</style>

### Basic
<Demo1 className="playground" />

---

### Import modules
<Demo2 className="playground" />

### A Third Example
<Demo1 className="playground" />
```

```ts
// stories/markdown-demo.jsx
import React from 'react'
import { Button } from './Button'
import { Playground } from '../src'
import Confetti from 'canvas-confetti'

export const Demo1 = ({ className }) => (
  <Playground
    className={className}
    code="<Button primary label={'hello'} />"
    providerProps={{
      scope: {
        Button,
      },
    }}
  />
)

export const Demo2 = ({ className }) => (
  <Playground
    className={className}
    code={`import Button from './Button';
import Confetti from 'canvas-confetti';

function cheer() {
  // left
  Confetti({ particleCount: 30, angle: 60, spread: 55, origin: { x: 0 } })

  // right
  Confetti({ particleCount: 30, angle: 120, spread: 55, origin: { x: 1 } })
}

() => <Button primary label={'Cheer !'} onClick={cheer} />
`}
    providerProps={{
      resolveImports: () => ({
        './Button': Button,
        'canvas-confetti': Confetti,
      }),
    }}
  />
)

export const Demo3 = ({ className }) => (
  <Playground
    className={className}
    code="<Button primary label={'hello'} />"
    providerProps={{
      scope: {
        Button,
      },
    }}
  />
)
```

**Conclusion:**

Like I said, a work-around exists. You can either use defaultExpanded (a bit verbose) or be careful to insert lots of line-breaks (a bit awkward). Otherwise, it is all-around a great tool. I will be happy to contribute a PR if you have an idea of how to fix this but don't have the time to implement it.

**Comment by hirohe** [comment]
thanks for reporting this problem!

it is had obvious issues of editor's expand/collapse when i use this in other project's storybook, i will try to fix it.

I run your example codes, the error stack shows the problem is [here](https://github.com/hirohe/storybook-addon-jarle-monaco/blob/main/src/components/Playground.tsx#L168), i think the useEffect loop caused the issue.

**Comment by hirohe** [comment]
I have released version [`0.5.0`](https://www.npmjs.com/package/storybook-addon-jarle-monaco), fixed the defaultExpanded and scrolling issues.

**Comment by andre-byrne** [comment]
Working like a charm! Thanks!!! I'll let you close the issue.
